uitools/widgets/Esri/ArcGISRuntime/Toolkit/BasemapGallery.cpp

```diff
@@ -0,0 +1,126 @@
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+#include "BasemapGallery.h"
+
+#include "ui_BasemapGallery.h"
+
+namespace Esri {
+namespace ArcGISRuntime {
+namespace Toolkit {
+
+ /*!
+  \class Esri::ArcGISRuntime::Toolkit::BasemapGallery
+  \inmodule EsriArcGISRuntimeToolkit
+  \ingroup ArcGISQtToolkitUiCppWidgetsViews
+  \since Esri.ArcGISRuntime 100.13
+  \brief The user interface for the BasemapGallery.
+  The BasemapGallery displays a collection of items representing basemaps from either ArcGIS Online, a user-defined portal,
+  or an array of Basemaps. When the user selects a basemap from the BasemapGallery, the  basemap rendered in the current
+  geoModel is removed from the given map/scene and replaced with the basemap selected in the gallery.
+  */
+
+ /*!
+  \brief Constructor.
+  \list
+    \li \a parent Parent widget.
+  \endlist
+
+  View mantains its associated controller, sets up the view itself, its model from the controller and 
+  connects listview clicked event to \l BasemapGallery::clickedItem.
+  \note geomodel should be manually set by calling \l setGeoModel.
+  */
+  BasemapGallery::BasemapGallery(QWidget* parent) :
+    QFrame(parent),
+    m_ui(new Ui::BasemapGallery),
+    m_controller(new BasemapGalleryController(this))
+  {
+    m_ui->setupUi(this);
+    GenericListModel* model = m_controller->gallery();
+    m_ui->listView->setModel(model);
+
+    //signal-slots
+    connect(m_ui->listView, &QListView::clicked, this, &BasemapGallery::clickedItem);
+  }
+
+  /*!
+  \brief Constructor with geomodel parameter.
+  \list
+    \li \a geomodel Geomodel.
+    \li \a parent Parent widget.
+  \endlist
+
+  Calls single QWidget constructor and sets the currentBasemap in the controller as the \a geomodel basemap passed. 
+  Sets the geomodel in the controller as \a geomodel.
+  */
+  BasemapGallery::BasemapGallery(GeoModel* geomodel, QWidget* parent) :
+    BasemapGallery(parent)
+  {
+    m_controller->setCurrentBasemap(geomodel->basemap());
+    m_controller->setGeoModel(geomodel);
+  }
+
+  /*!
+    \brief Destructor.
+   */
+  BasemapGallery::~BasemapGallery()
+  {
+    delete m_ui;
+  }
+
+  /*!
+    \brief Returns the controller.
+   */
+  BasemapGalleryController* BasemapGallery::controller() const
+  {
+    return m_controller;
+  }
+
+  /*!
+    \brief Setting the controller currentbasemap and geomodel.
+    \a geomodel Geomodel
+   */
+  void BasemapGallery::setGeoModel(GeoModel* geomodel)
+  {
+    if (geomodel != nullptr)
+    {
+      m_controller->setCurrentBasemap(geomodel->basemap());
+      m_controller->setGeoModel(geomodel);
+    }
+  }
+
+/*!
+    \brief Getter of current \c GeoModel .
+    returns \c GeoModel stored in the model managed by the controller.
+   */
+  GeoModel* BasemapGallery::getCurrentGeoModel()
```

Comment:
:x: Typically, Qt-style getters/setters should be of the form:

```c++
Foo foo();
void setFoo(Foo foo);
```

So, drop the `getCurrent` and rename this method to `geoModel`.

Reply:
So they should not based on the property name but rather on the type name

Reply:
There's a `currentBasemap` property. Are you confusing the `geoModel` property with that? 

The `currentBasemap` is derived from the active basemap on the `GeoModel`.